src/transport_performance/gtfs/gtfs_utils.py

```diff
@@ -343,21 +351,22 @@ def filter_gtfs_around_trip(
         An error is raised if a shapeID is not available
 
     """
-    # TODO: Add datatype defences once merged
+    # NOTE: No defence for units as its deleted later on
+    _gtfs_defence(gtfs, "gtfs")
+    _type_defence(trip_id, "trip_id", str)
+    _type_defence(buffer_dist, "buffer_dist", int)
+    _type_defence(crs, "crs", str)
+    _check_parent_dir_exists(out_pth, "out_pth", create=True)
     trips = gtfs.feed.trips
     shapes = gtfs.feed.shapes
 
     shape_id = list(trips[trips["trip_id"] == trip_id]["shape_id"])[0]
 
     # defence
-    # try/except for math.isnan() returning TypeError for strings
-    try:
-        if math.isnan(shape_id):
-            raise ValueError(
-                "'shape_id' not available for trip with trip_id: " f"{trip_id}"
-            )
-    except TypeError:
-        pass
+    if pd.isna(shape_id):
+        raise ValueError(
+            "'shape_id' not available for trip with trip_id: " f"{trip_id}"
+        )
```

Comment:
This relates to #180 :
"3. 1 Use pd.isna() in filter_gtfs_around_trip() - handles strings, no need to try/except."

Straight forward  implementation, though I don’t see a test for this. As it's a public function probably for the best. Presume you would need to have a GTFS fixture  with a valid trip ID but no  corresponding shape_id. Take one of our fixtures and lookup a shape_id for a given ID and remove it prior to testing this feature.

Reply:
![image](https://github.com/datasciencecampus/transport-network-performance/assets/121952297/dfadc33d-742b-442e-8b13-fd46bdb7f9e2)
Already implemented. Not covered in the scope of this PR since this is changing previous functionality that does the same thing (test was already written).